GLMakie/src/drawing_primitives.jl

```diff
@@ -214,32 +214,25 @@ function draw_atomic(screen::GLScreen, scene::Scene, @nospecialize(x::Union{Scat
         # signals not supported for shading yet
         gl_attributes[:shading] = to_value(get(gl_attributes, :shading, true))
         marker = lift_convert(:marker, pop!(gl_attributes, :marker), x)
+        
+        positions = handle_view(x[1], gl_attributes)
+        positions = apply_transform(transform_func_obs(x), positions)
+        
         if isa(x, Scatter)
-            markerspace = get(gl_attributes, :markerspace, :pixel)
+            space = get(gl_attributes, :space, :data)
+            mspace = get(gl_attributes, :markerspace, :pixel)
             cam = scene.camera
-            positions = map(
-                    x[1], get(gl_attributes, :space, :data), markerspace, 
-                    transform_func_obs(x), cam.projectionview, cam.resolution
-                ) do positions, space, markerspace, tf, _, _
-
-                mat = Makie.clip_to_space(cam, markerspace) * Makie.space_to_clip(cam, space)
-                map(positions) do pos
-                    p = apply_transform(tf, pos)
-                    p4d = mat * Makie.to_ndim(Point4f, Makie.to_ndim(Point3f, p, 0), 1)
-                    p4d[SOneTo(3)] / p4d[4]
-                end
+            gl_attributes[:preprojection] = map(space, mspace, cam.projectionview) do space, mspace, pv
```

Comment:
that looks a bit like this is rather the full projection matrix? Or is there some other projection matrix applied afterward?

Reply:
It's split up to go from space -> markerspace via `preprojection` and from markerspace to clip space via `projection * view`. Basically `preprojection` does what 
https://github.com/JuliaPlots/Makie.jl/blob/85a6d145d3d931ebde190046d20482f6eb6bcc69/src/layouting/layouting.jl#L238
did without the `transform` and from a variable space to a variable markerspace (instead of just data -> pixel (or data))